src/ledger/service.ts

```diff
@@ -0,0 +1,190 @@
+import TransportWebUSB from '@ledgerhq/hw-transport-webusb';
+import WavesLedger from '@waves/ledger';
+import { Account } from 'accounts/types';
+import Background from 'ui/services/Background';
+import { LedgerSignRequest } from './types';
+
+export enum LedgerServiceStatus {
+  Disconnected = 'DISCONNECTED',
+  UsedBySomeOtherApp = 'USED_BY_SOME_OTHER_APP',
+  Ready = 'READY',
+}
+
+function delay(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+class LedgerService {
+  private _connectionRetryIsNeeded: boolean;
+  private _ledger: WavesLedger;
+  private _signRequestPromise = Promise.resolve();
+  private _status = LedgerServiceStatus.Disconnected;
+
+  get ledger() {
+    return this._ledger;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  async connectUsb(networkCode: string) {
+    await this.disconnect();
+
+    this._ledger = new WavesLedger({
+      debug: true,
+      openTimeout: 3000,
+      listenTimeout: 30000,
+      exchangeTimeout: 30000,
+      networkCode: networkCode.charCodeAt(0),
+      transport: TransportWebUSB,
+    });
+
+    while (this._ledger && this._status !== LedgerServiceStatus.Ready) {
+      await this.updateStatus();
+
+      if (this._connectionRetryIsNeeded) {
+        await delay(1000);
+        continue;
+      }
+    }
+  }
+
+  async updateStatus() {
+    this._connectionRetryIsNeeded = false;
+
+    if (!this._ledger) {
+      return;
+    }
+
+    try {
+      if (await this._ledger.probeDevice()) {
+        this._status = LedgerServiceStatus.Ready;
+      } else {
+        this._connectionRetryIsNeeded = true;
+      }
+    } catch (err) {
+      if (err instanceof Error) {
+        if (
+          /No device selected|device was disconnected|user gesture to show a permission request/i.test(
+            err.message
+          )
+        ) {
+          this.disconnect();
+        } else if (/Unable to claim interface/i.test(err.message)) {
+          this.disconnect(LedgerServiceStatus.UsedBySomeOtherApp);
+        } else if (
+          /An operation that changes the device state is in progress/i.test(
+            err.message
+          )
+        ) {
+          this._connectionRetryIsNeeded = true;
+        } else {
+          console.error('NO MATCH FOR ERROR', err);
+        }
+      } else {
+        console.error('NON-ERROR THROWN', err);
+      }
```

Comment:
Should Sentry catch these errors or is it just to be ignored?

Reply:
No, now sentry won't catch them, but I'll fix it